### DISCUSSION TITLE
EIM for part of a LincombOperator

### DISCUSSION THREAD

**Discussion by niklasreich** [pymor/pymor#2199] [Q&A] [answered]
I have a full order model `fom` with one operator that is a LincombOperator, which itself includes multiple operators in its attribute `operators`. I want to apply the EIM only to the one operator within my linear combination that is nonlinear.
To do this I am trying to use the function `interpolate_operators()`. This function already has a parameter `operator_names` but I am struggling to pass only part of my linear combination (so something like `["operator.operators[2]"]` if `fom.operator.operators[2]` is the desired operator) to the function, because the `getattr()` function within `interpolate_operators()` says this is not an attribute of my `fom` anymore.
How can I solve this?

**Comment by sdrave** [comment, marked as answer]
Yes, `interpolate_operators` is not smart enough to only treat parts of operators. I'm not sure if it's worth the effort to add that feature. If you strip the features that you don't need, `interpolate_operators` reduces to like 10 lines of code. So I would recommend to copy-paste the relevant parts and adapt them to handle the desired operator. 

Something like:

```python
op = fom.operator.operators[2]
evaluations = op.range.empty()
for mu in parameter_sample:
    U = fom.solve(mu)
    evaluations.append(op.apply(U, mu=mu)
dofs, basis, data = ei_greedy(evaluations, copy=False, **further_options)
ei_op = EmpiricalInterpolatedOperator(op, dofs, basis, triangular=True)  #False for DEIM
new_ops = [ei_op if i == 2 else op for i, op in enumerate(fom.operator.operators)]
ei_fom = fom.with_(operator=fom.operator.with_(operators=new_ops))
```

**Reply by niklasreich** [reply]
Thanks for the insight, I will do that. I just wanted to make sure that this isn't a feature that is already included and I am just not calling the function in the correct way.
